syncer/syncer.go

```diff
@@ -1316,7 +1348,7 @@ func (s *Syncer) syncDML(
 			time.Sleep(time.Duration(t) * time.Second)
 		})
 		// use background context to execute sqls as much as possible
-		ctctx, cancel := tctx.WithContext(context.Background()).WithTimeout(maxDMLExecutionDuration)
+		ctctx, cancel := tctx.WithTimeout(maxDMLExecutionDuration)
```

Comment:
So what about add a wg.wait when find isTransactionEnd before do `runCancel`. Or we should rollback to use the context.Background to handle this situation.

Reply:
I think both SGTM.

Reply:
> Suggest using background context here. To avoid being affected by other contexts.
> If `isTransactionEnd` is true and we directly run `runCancel`, the safePoint and new global checkpoint won't be saved to the downstream database.

Are you talking about this problem?

https://github.com/pingcap/dm/pull/1928/#discussion_r684871196

Reply:
Maybe just revert this change and use the origin background ctx can make things a little easier @okJiang

Reply:
I think add a `jobwg.Wait` before `runCancel()` is enough.